otherlibs/unix/select.c

```diff
@@ -47,26 +47,25 @@ static int fdlist_to_fdset(value fdlist, fd_set *fdset, int *maxfd)
 
 static value fdset_to_fdlist(value fdlist, fd_set *fdset)
 {
-  value l;
-  value res = Val_int(0);
+  CAMLparam0();
+  CAMLlocal2(l, res);
 
-  Begin_roots2(l, res);
-    for (l = fdlist; l != Val_int(0); l = Field(l, 1)) {
-      int fd = Int_val(Field(l, 0));
-      if (FD_ISSET(fd, fdset)) {
-        value newres = caml_alloc_small(2, 0);
-        Field(newres, 0) = Val_int(fd);
-        Field(newres, 1) = res;
-        res = newres;
-      }
+  for (l = fdlist; l != Val_int(0); l = Field(l, 1)) {
```

Comment:
Apropos https://github.com/ocaml/ocaml/pull/11002/files#r804270079:
```suggestion
  for (l = fdlist; l != Val_emptylist; l = Field(l, 1)) {
```

Reply:
Fixed (and also changed a similar `Val_int(0)` in the same file).